oecophylla/taxonomy/parser.py

```diff
@@ -0,0 +1,55 @@
+import os
+import pandas as pd
+
+
+def combine_bracken(dir_bracken_reports):
+    """Combines bracken counts for several samples into one table.
+
+    Parameters
+    ----------
+    dir_bracken_reports : str
+        Pathname of directory that contains bracken abundance estimation files
+        per samples.
+
+    Returns
+    -------
+    Pandas.DataFrame with rows for features, columns for samples.
+    Feature labels are NCBI taxonomy IDs.
+
+    Notes
+    -----
+    We don't know which files in the given directory are bracken abundance
+    estimation files.
+    Therefore, we iterate over all files and try to a) parse the file as a tab
+    separated table and b) check if the second column is named 'taxonomy_id'.
+    More information about the file format:
+    https://ccb.jhu.edu/software/bracken/index.shtml?t=manual
+    """
+    samples = []
+    # walking over the given directory.
+    # We don't know which files are actual bracken abundance estimation files.
+    for _file in next(os.walk(dir_bracken_reports))[2]:
+        try:
+            sample_abund = pd.read_csv(dir_bracken_reports + '/' + _file,
```

Comment:
this will not be fun if the directory happens to contain some large files. is there a better test? like reading the first 100 bytes?

Reply:
I am more worried about many small files and therefore I wanted to limit the number of file accesses. In principle, I could open the file, read the first line and if it matches the expected header open the file again through pandas.read_csv. But that is then two file reads :-/

What do you think about reading the file line by line, writing results into a string, break if first line does not match and otherwise pass string to pandas.read_csv as a StringIO?

Reply:
what about:

```python
import itertools

...

f = open(_file)
line = f.readline()
if not some_test(line):
    f.close()
    continue
file_contents = itertools.chain([line], f)
sample_abund = pd.read_csv(file_contents, ...)
```

It might need some wrangling to get `file_contents` into something that `pd.read_csv` will be happy with